### DISCUSSION TITLE
How to run bcc tools in a script instead of one-line commands

### DISCUSSION THREAD

**Discussion by divija95** [bpftrace/bpftrace#1960] [Q&A]
Hi, I plan to measure various OS latencies through bcc/bpftrace tools for different workloads. Please let me know how to achieve this through a script?

**Comment by danobi** [comment]
`sudo bpftrace $FILENAME`?

**Reply by divija95** [reply]
But I would like to run the tool right before launching a workload. Similarily, I want to terminate the tool once the workload completes. I am not sure how to acheive all this through a shell script

**Reply by fbs** [reply]
bpftrace has the `-c` flag which runs a child command. 

Or you could do something like

```
#!/bin/bash

bpftrace myscript.bt &

mycommand >/dev/null 2>&1 &

wait %2

kill -SIGINT %1
```

**Comment by fbs** [comment]
If you mean a bash script you could do something like

```
#!/bin/bash
$var = banana
(
cat << EOF
kprobe:$var { @=5; }
EOF
) | bpftrace -
```
